src/hooks/usePageReducer.js

```diff
@@ -37,25 +38,24 @@ import { debounce } from '../utilities/index';
  *
  * @param {String} page                   routeName for the current page.
  * @param {Object} initialState           Initial state of the reducer
- * @param {Func}   initializer            Function to generate the initial state (optional)
+ * @param {Func}   initialiser            Function to generate the initial state (optional)
```

Comment:
8dc05cd9142146d6a373e0177c940c551157e1e8.

My opinion if anyone interested: I think that having an initialiser parameter leaves the usage a bit ambiguous, as it implies the caller has to have knowledge of what pages the reducer knows about or always provide a backup initialiser. I think it'd be nice to extract out all initialisation from the caller by essentially restricting it to callers which have hardcoded initialisers (can think of these as the set of pages the function is defined over). It seems counter-intuitive, but I think that adding the optional parameter actually makes the reducer more as opposed to less tightly coupled.

Reply:
more tightly coupled with what?

The intention with this was essentially to create a middleware for `useReducer`.. it slowly expanded into injecting props for a bit of DRYness (all of them could just be called in a page). 

I think possibly the best option would be to make a `useThunkReducer` hook - which is the middleware hook. Then, the `usePageReducer` hook could just inject properties. However, I also wonder now if we should just put the field injections straight into the initial state - maybe that would be less confusing heh

Reply:
Oops, just saw this. I agree with that.

My anti-initialisation parameter thoughts clarified:

I like the idea of `usePageReducer` only being usable by pages which are known to the application, as it allows all the state to be abstracted away from the page.  It feels a bit wrong for the caller to be initialising state which it doesn't maintain.

Thinking of DRY, it also makes sense that all pages should have the source of truth for their initialisers in a single place, rather than having a division between pages with and without hardcoded initialisers.

In that context, I think your suggestion of separating out `usePageReducer` into two reducer is a good one, it allows a clearer separation between the more static "page" state and the more dynamic/side-effecty "data" state.